python/lsst/rapid/analysis/quickLook.py

```diff
@@ -0,0 +1,281 @@
+        isrConfig.doDark = False
+        isrConfig.doFlat = False
+        isrConfig.doFringe = False
+        isrConfig.doDefect = False
+        isrConfig.doLinearize = False
+        isrConfig.doCrosstalk = False
+        isrConfig.doBrighterFatter = False
+        isrConfig.usePtcGains = False
+
+        if bias:
+            isrConfig.doBias = True
+            self.log.info("Running with bias correction")
+
+        if dark:
+            isrConfig.doDark = True
+            self.log.info("Running with dark correction")
+
+        if flat:
+            isrConfig.doFlat = True
+            self.log.info("Running with flat correction")
+
+        # TODO: deal with fringes here
+        if defects:
+            isrConfig.doDefect = True
+            self.log.info("Running with defect correction")
+
+        if linearizer:
+            isrConfig.doLinearize = True
+            self.log.info("Running with linearity correction")
+
+        if crosstalk:
+            isrConfig.doCrosstalk = True
+            self.log.info("Running with crosstalk correction")
+
+        if bfKernel:
+            isrConfig.doBrighterFatter = True
+            self.log.info("Running with brighter-fatter correction")
+
+        if ptc:
+            isrConfig.usePtcGains = True
+            self.log.info("Running with ptc correction")
+
+        isrConfig.doWrite = False
+        isrTask = IsrTask(config=isrConfig)
+        result = isrTask.run(ccdExposure,
+                             camera=camera,
+                             bias=bias,
+                             dark=dark,
+                             flat=flat,
+                             #  fringes=pipeBase.Struct(fringes=None),
+                             defects=defects,
+                             linearizer=linearizer,
+                             crosstalk=crosstalk,
+                             bfKernel=bfKernel,
+                             bfGains=bfGains,
+                             ptc=ptc,
+                             crosstalkSources=crosstalkSources,
+                             isGen3=isGen3,)
+
+        postIsr = result.exposure
+
+        if self.config.doRepairCosmics:
+            try:  # can fail due to too many CRs detected, and we always want an exposure back
+                self.log.info("Repairing cosmics...")
+                if postIsr.getPsf() is None:
+                    installPsfTask = InstallGaussianPsfTask()
+                    installPsfTask.run(postIsr)
+
+                repairConfig = CharacterizeImageTask.ConfigClass()
+                repairConfig.doMeasurePsf = False
+                repairConfig.doApCorr = False
+                repairConfig.doDeblend = False
+                repairConfig.repair.cosmicray.nCrPixelMax = 200000
+                repairTask = CharacterizeImageTask(config=repairConfig)
+
+                repairTask.repair.run(postIsr)
+            except Exception as e:
+                self.log.warn(f"During CR repair caught: {e}")
```

Comment:
This feels like it should be an `error`? Also, `warn` is deprecated: use `warning` instead.

Reply:
This is probably catching the "Too many CR" exception that can arise easily with images that have not had all the ISR steps run (most commonly if defects are unavailable).  Treating this as a warning means processing continues with an image that just retains the cosmic rays.

Reply:
This should not be an error, no. The whole point of this task is that it _always_ gives you back an image (as long as there is a raw for that dataId, of course). Just like how you don't apply a bias if you can't find one rather than failing, you just don't apply CR interpolation if it fails, you simply return the image anyway. But unlike isr stuff, you can't know till you try, all you can do is catch the error.

Reply:
It's actually not because of what Chris says in practice though, usually (though it probably could be too), it's actually that high flux flats trigger _tons_ of CRs. I think it's a problem with our saturation levels, but I've never had time to dig into it. This does remind me that I should ticket it though.

Reply:
DM-33862